pypsa/components.py

```diff
@@ -891,11 +891,11 @@ def consistency_check(self):
 
                 # check for NaN values:
                 if max_pu.isnull().values.any():
-                    for col in max_pu.dropna(axis=1, how='any').columns:
+                    for col in max_pu.columns[max_pu.isna().any()].tolist():
```

Comment:
please stick with `isnull`. `isna` and `isnull` are synonymous, let's keep it consistent.

you don't need the `tolist` instantiation: `for col in max_pu.columns[max_pu.isna().any()]:` works fine.